### PR TITLE
Use $real_pkgname for pip uninstall command

### LINKED ISSUE #606: Pip uninstall does not support pkgname variable
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 5.4.0
- Ruby: ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-linux-gnu]
- Distribution: Ubuntu 18.04
- Module version: 5.0, 6.0

## How to reproduce (e.g Puppet code you use)
```puppet
        python::pip { '/opt/virtualenv/oco3/wheel':
            ensure     => 'absent',
            pkgname    => 'wheel',
            virtualenv => '/opt/virtualenv/oco3'
        }
```

## What are you seeing
Puppet error on executing pip uninstall

## What behaviour did you expect instead
Uninstalled package determined by `$pkgname` within given `$virtualenv`

## Output log
Relevant part of the output
```
Python::Pip[/opt/virtualenv/oco3/wheel]/Exec[pip_uninstall_/opt/virtualenv/oco3/wheel]/returns: ERROR: Invalid requirement: '/opt/virtualenv/oco3/wheel'
Python::Pip[/opt/virtualenv/oco3/wheel]/Exec[pip_uninstall_/opt/virtualenv/oco3/wheel]/returns: Hint: It looks like a path. File '/opt/virtualenv/oco3/wheel' does not exist.
Error: 'echo y | /opt/virtualenv/oco3/bin/pip uninstall   /opt/virtualenv/oco3/wheel' returned 1 instead of one of [0]
Error: Python::Pip[/opt/virtualenv/oco3/wheel]/Exec[pip_uninstall_/opt/virtualenv/oco3/wheel]/returns: change from 'notrun' to ['0'] failed: 'echo y | /opt/virtualenv/oco3/bin/pip uninstall   /opt/virtualenv/oco3/wheel' returned 1 instead of one of [0]
```

## Any additional information you'd like to impart
BUG location https://github.com/voxpupuli/puppet-python/blob/master/manifests/pip.pp#L223
```
      default: {
        # Anti-action, uninstall.
        $command        = "echo y | ${pip_env} uninstall ${uninstall_args} ${proxy_flag} ${name}"
        $unless_command = "! ${pip_env} list | grep -i -e '${grep_regex}'"
      }
```
As you can see this piece of code doesn't respect `$pkgname` from define params. Instead it uses `$name` to build the command

### PR DESCRIPTION
**Pull request by brabiega**
Fix #606

#### Pull Request (PR) description
Use $real_pkgname for pip uninstall command

#### This Pull Request (PR) fixes the following issues
Fixes #606

### CODE REVIEW

**Review by kenyon — APPROVED** [review]
Would be good to also add a test for this so that it doesn't break again.

**Comment by brabiega** [comment]
I've added the test, should be enough.

File: manifests/pip.pp [hunk 1]
```diff
@@ -220,7 +220,7 @@
 
       default: {
         # Anti-action, uninstall.
-        $command        = "echo y | ${pip_env} uninstall ${uninstall_args} ${proxy_flag} ${name}"
+        $command        = "echo y | ${pip_env} uninstall ${uninstall_args} ${proxy_flag} ${real_pkgname}"
```
Comment (bastelfreak) [inline manifests/pip.pp:223]: thanks for the PR! Can you take a look at the acceptance tests in spec/acceptance/ and add one there that reflects this change? If you need any help with it feel free to join our IRC channel #voxpupuli on Freenode our or slack channel #voxpupuli on https://slack.puppet.com/

File: spec/acceptance/pip_spec.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+      python::pip { 'agent package':
+        pkgname => 'agent',
+        ensure  => '0.1.2',
+      }
+      PUPPET
+
+      apply_manifest(pp, catch_failures: true)
+      apply_manifest(pp, catch_changes: true)
+  end
```
Comment (bastelfreak) [inline spec/acceptance/pip_spec.rb:20]: I think between line 19 and 20 is one `end` missing?

Comment (brabiega) [reply spec/acceptance/pip_spec.rb:20]: Mostlikely, I'll check in the evening. Is there a way to run the acceptance tests locally?

**Comment by brabiega** [comment]
I guess system wide pip path may differ on different OSes.
`/usr/bin/pip` - this is not universal.

I'll fix the acceptance tests to run pip inside virtualenv.

**Pull request merged**